rustls-mbedcrypto-provider/src/kx.rs

```diff
@@ -28,34 +30,39 @@ use rustls::ffdhe_groups;
 use rustls::ffdhe_groups::FfdheGroup;
 use rustls::Error;
 use rustls::NamedGroup;
-/// A key-exchange group supported by *mbedtls*.
+/// An EC key-exchange group supported by *mbedtls*.
 ///
 /// All possible instances of this type are provided by the library in
 /// the `ALL_KX_GROUPS` array.
-struct KxGroup {
+pub struct KxGroup<T: RngCallback> {
     /// The IANA "TLS Supported Groups" name of the group
-    name: NamedGroup,
+    pub name: NamedGroup,
 
-    /// The corresponding [`agreement::Algorithm`]
-    agreement_algorithm: &'static agreement::Algorithm,
+    /// The corresponding agreement algorithm
+    pub agreement_algorithm: &'static agreement::Algorithm,
+
+    /// Function to provider a RNG
```

Comment:
Nit: I suggest `callback to produce RNGs when needed` for the doc comment, and `rng_provider` for the field name.